### ISSUE TITLE
File and Line number is reported as coming from bunyan instead of location of log.

### ISSUE THREAD

**Issue by jduprey** [philmander/browser-bunyan#1]
I'm using browser bunyan and I've configured it to log the source file and line number, but it is reporting file and line in bunyan.

Initialized:
```
$provide.decorator('$log', function($delegate) {
      $delegate = bunyan.createLogger({
        name: 'tracer-logger',
        src: true,
        streams: [{
          level: 'info',
          stream: new bunyan.ConsoleFormattedStream(),
          type: 'raw'
        }]
      });
      return $delegate;
    });
```

Console log:
`[20:01:30:0512]  INFO: tracer-logger: Websocket batch update [23].        browser-bunyan.min.js:6 `

https://www.evernote.com/l/AAIdfssu8jtKRJlPsailB1hqB82VVadfzGM

What am I doing wrong?

**Comment by RanzQ** [comment]
@jduprey This is a limitation with the default ConsoleFormattedStream. Browser outputs the line where `console.log` is called, but the real info is in `rec.src`. You can create your own stream (check what [ConsoleFormattedStream does](https://github.com/philmander/browser-bunyan/blob/master/lib/bunyan.js#L164)). Since this is not the full node-bunyan, some things might not work as expected. :)

**Comment by philmander** [comment]
Thankyou @RanzQ 

I'm looking into this a bit more.

**Comment by philmander** [comment]
@RanzQ I tried to replicate your fix, but I get an error: 

`browser-bunyan.js:99 Uncaught TypeError: Object.defineProperty called on non-object`

This is also after setting `src:true` in the logger options. Does it definitely work for you?

This is probably why I commented out that line originally. I'm thinking I might need to do something along these lines:

```
var err = new Error();
var stack = err.stack;
rec.src = getThirdCallerFromStack(stack) // I think this part would require some x-browser stack trace parsing code
```

**Comment by RanzQ** [comment]
@philmander Yeah sorry, forgot to update the answer. After uncommenting the line I ended up with the same error. My "fix" was just for the string format.

I can try with original bunyan if it works with webpack.

**Comment by RanzQ** [comment]
So, wasn't able to make it work with the node-bunyan either, I just get an empty object from [getCaller3Info](https://github.com/philmander/browser-bunyan/blob/master/lib/bunyan.js#L111).

I've found it best to debug with console.log and create the final logger with browser-bunyan. When using named child loggers you don't necessarily need the line number. :)

**Comment by philmander** [comment]
I've added line number support [on this branch](https://github.com/philmander/browser-bunyan/tree/log-line-numbers)

I will do some testing/add-support in other browsers, probably IE and Safari and then merge and then release

**Comment by philmander** [comment]
I've released a fix working across major browsers @ 0.3.0

**Comment by jduprey** [comment]
Thank you!
